### PR TITLE
New ordering of contribution steps for nf-core modules

### PR DESCRIPTION
**Pull request by Krannich479**
- new ordering of the steps when contributing to nf-core modules
- cheat sheet description in Step 1 for an all-in-one conda env of dependencies

@netlify /docs/contributing/modules

### CODE REVIEW

File: src/content/docs/contributing/modules.md [hunk 1]
```diff
@@ -49,17 +49,25 @@ If the module doesn't exist on `nf-core/modules`:
 
 We have implemented a number of commands in the `nf-core/tools` package to make it incredibly easy for you to create and contribute your own modules to nf-core/modules.
 
-1. Install the latest version of [`nf-core/tools`](https://github.com/nf-core/tools#installation) (`>=2.7`)
+1. Install any of [`Docker`](https://docs.docker.com/engine/installation/), [`Singularity`](https://www.sylabs.io/guides/3.0/user-guide/) or [`Conda`](https://conda.io/miniconda.html)
+   <details>
+     <summary> 🛠️ Conda environment for the Module contribution workflow </summary>
+     If you use the conda package manager you can install all dependencies for the New module workflow directly into a new environment via
+     <code>
+       conda create -n nf-core -c bioconda "nextflow>=21.04.0" "nf-core>=2.7" nf-test prettier
```
Comment (mashehu) [inline src/content/docs/contributing/modules.md:57]:
```suggestion
       conda create -n nf-core -c bioconda "nextflow>=21.04.0" "nf-core>=2.7" nf-test pre-commit
```

File: src/content/docs/contributing/modules.md [hunk 2]
```diff
@@ -49,17 +49,25 @@ If the module doesn't exist on `nf-core/modules`:
...
+       conda activate nf-core
+     </code>
+   </details>
 2. Install [`Nextflow`](https://www.nextflow.io/docs/latest/getstarted.html#installation) (`>=21.04.0`)
```
Comment (mashehu) [inline src/content/docs/contributing/modules.md:61]: needs to be rewritten, so one either uses conda with the hint above or the manual steps. maybe something similar to https://multiqc.info/docs/getting_started/installation/

File: src/content/docs/contributing/modules.md [hunk 3]
```diff
@@ -49,17 +49,25 @@ If the module doesn't exist on `nf-core/modules`:
...
-4. Install any of [`Docker`](https://docs.docker.com/engine/installation/), [`Singularity`](https://www.sylabs.io/guides/3.0/user-guide/) or [`Conda`](https://conda.io/miniconda.html)
-5. Setup up [pre-commit](https://pre-commit.com/) (comes packaged with [`nf-core/tools`](https://github.com/nf-core/tools#installation), watch the [pre-commit bytesize talk](https://www.youtube.com/watch?v=08d6zv6zvdM&t=215) if you want to know more about it) to ensure that your code is linted and formatted correctly before you commit it to the repository
+3. Install the latest version of [`nf-core/tools`](https://github.com/nf-core/tools#installation) (`>=2.7`)
+4. Install [`nf-test`](https://code.askimed.com/nf-test/installation/)
+5. [Fork and clone the nf-core/modules repo locally](#uploading-to-nf-coremodules)
+6. Setup up [pre-commit](https://pre-commit.com/) (comes packaged with [`nf-core/tools`](https://github.com/nf-core/tools#installation), watch the [pre-commit bytesize talk](https://www.youtube.com/watch?v=08d6zv6zvdM&t=215) if you want to know more about it) to ensure that your code is linted and formatted correctly before you commit it to the repository
```
Comment (mashehu) [inline src/content/docs/contributing/modules.md:65]: why are you flipping step 6 with 5?

Comment (ewels) [reply src/content/docs/contributing/modules.md:65]: That's the point of the PR. See [slack thread](https://nfcore.slack.com/archives/CJRH30T6V/p1710758990571599). You can't run `pre-commit install` if there's no project directory to run it in.

Comment (mashehu) [reply src/content/docs/contributing/modules.md:65]: ah, didn't see that. Good point then!

File: src/content/docs/contributing/modules.md (hunk 1 above)
Comment (Krannich479) [reply src/content/docs/contributing/modules.md:57]: `pre-commit` comes for free with nf-core. `prettier` has to be installed separately.

File: src/content/docs/contributing/modules.md (hunk 2 above)
Comment (Krannich479) [reply src/content/docs/contributing/modules.md:62]: Thanks for the pointer. I'll take a look at this.

**Comment by Krannich479** [comment]
For the sake of simplicity I deleted my conda suggestion. Now this PR is purely about the re-ordering of the steps. 
However, the "Lint code / pre-commit" check still fails. I inspected the erro log and it seems the CI is checking a ton of files that I haven't even touched. Can you guys tell what's going on there? @ewels @mashehu

**Comment by mashehu** [comment]
sorry for the mess in the CI. An automated PR somehow circumvented some of the prettier checks and added incorrectly formatted code to main (we always test all files for simplicity and completeness sake, that's why they come up in your PR. I like your tip with the conda setup, will add it as a [collapsible admonition](https://nf-co.re/docs/contributing/website_markdown#collapsing-admonitions).

**Comment by mashehu** [comment]
@nf-core-bot fix linting

**Comment by Krannich479** [comment]
tip_box

@mashehu addition much appreciated. 👍 
This is what it looks on my side rn.

File: src/content/docs/contributing/modules.md [hunk 4]
```diff
@@ -49,17 +49,29 @@ If the module doesn't exist on `nf-core/modules`:
 
 We have implemented a number of commands in the `nf-core/tools` package to make it incredibly easy for you to create and contribute your own modules to nf-core/modules.
 
-1. Install the latest version of [`nf-core/tools`](https://github.com/nf-core/tools#installation) (`>=2.7`)
+1. Install any of [`Docker`](https://docs.docker.com/engine/installation/), [`Singularity`](https://www.sylabs.io/guides/3.0/user-guide/) or [`Conda`](https://conda.io/miniconda.html)
+
+:::tip{title="Single step conda installation" collapse}
+If you use the conda package manager you can setup a new environment and install all dependencies for the new module workflow in one step with:
```
Comment (mashehu) [inline src/content/docs/contributing/modules.md:55]:
```suggestion

If you use the conda package manager you can setup a new environment and install all dependencies for the new module workflow in one step with:
```

**Review by mashehu — APPROVED** [review]
Thanks for getting that last error!

**Comment by Krannich479** [comment]
Tbh I think it was just the ways it's displayed on GH. The website has all the right formatting. Sorry for the confusion. I think this PR is ready to be merged.

**Pull request merged**